src/screens/welcome/Welcome.tsx

```diff
@@ -51,13 +52,21 @@ export const Welcome = ({ route }: AuthNavigationProps<'WELCOME'>) => {
     )
   }
 
+  useEffect(() => {
+    const getIsUserLoggedIn = async () => {
+      const userSeenDashboard = await getItem('seenDashboard')
+      setHasUserSeenDashboard(!!userSeenDashboard)
+    }
+    getIsUserLoggedIn()
+  }, [])
```

Comment:
we have a useAsyncEffect hook. The regular useEffect looks good to me, but you can try it out, as it may look even better